### PR TITLE
Dependabot clearout

### PR DESCRIPTION
**Pull request by cgsunkel**
## Description of change

I've noticed we had a few open Dependabot PRs where tests known to be flaky hadn't been re-run, and a couple more with easy-to-fix breaking changes. These tests are now passing so we can upgrade these dependencies.

## Checklist

[//]: # 'When submitting a PR make sure the code review guidelines have been satisfied.
https://github.com/uktrade/data-hub-frontend/blob/main/docs/Code%20review%20guidelines.md'

- [ ] Has the branch been rebased to main?
- [ ] Automated tests (Any of the following when applicable: Unit, Functional or End-to-End)
- [ ] Manual compatibility testing (Browsers: Chrome, Firefox, Edge, Safari)

### CODE REVIEW

File: test/a11y/cypress/config/urlTestExclusions.js [hunk 1]
```diff
@@ -69,6 +69,7 @@ export const urlTestExclusions = [
   { url: '/reminders/settings/my-tasks-task-completed/' },
   { url: '/export/' },
   { url: '/tasks/:taskId/edit' },
+  { url: '/companies/:companyId/overview' },
```
Comment (cgsunkel) [inline test/a11y/cypress/config/urlTestExclusions.js:72]: This test is incredibly flaky, so this page has been skipped until we start looking into these failures.

File: src/client/provider.jsx [hunk 2]
```diff
@@ -25,7 +25,7 @@ const history = createBrowserHistory()
 
 const store = configureStore({
   devTools: process.env.NODE_ENV === 'development',
-  middleware: [sagaMiddleware, routerMiddleware(history)],
+  middleware: () => [sagaMiddleware, routerMiddleware(history)],
```
Comment (cgsunkel) [inline src/client/provider.jsx:28]: The latest versions of `@reduxjs/toolkit` require this to be a callback

File: cypress.visual.config.js [hunk 3]
```diff
@@ -3,7 +3,6 @@ require('dotenv').config()
 
 module.exports = defineConfig({
   video: true,
-  videoUploadOnPasses: false,
```
Comment (cgsunkel) [inline cypress.visual.config.js:6]: This `videoUploadOnPasses` option has been deprecated by Cypress

File: .github/dependabot.yml [hunk 4]
```diff
@@ -22,6 +22,9 @@ updates:
   - dependency-name: formidable 
     versions:
       - "> 2.1.2"
+  - dependency-name: webpack-dev-server # Waiting for Cypress to add support
+    versions:
+      - "> 4.15.1"
```
Comment (cgsunkel) [inline .github/dependabot.yml:27]: We can't upgrade this dependency until Cypress start supporting it - https://github.com/cypress-io/cypress/issues/28924

File: Dockerfile.dependencies [hunk 5]
```diff
@@ -53,7 +53,7 @@ COPY --chown=node:node package*.json "$HOME/"
 WORKDIR $HOME
 
 # Install specific version of cypress
-RUN npm install -g cypress@12.17.4 \
+RUN npm install -g cypress@13.6.6 \
```
Comment (oliverjwroberts) [inline Dockerfile.dependencies:56]: You're probably already on it, but just to flag you'll need to push an updated dependencies image to GCR (see `Docker.md` in the docs) and update the version used within the `Dockerfile` and `Dockerfile.storybook`. 

Happy to support if needed.

**Review by oliverjwroberts — CHANGES_REQUESTED** [review]

File: Dockerfile.dependencies (hunk 5 above)
Comment (cgsunkel) [reply Dockerfile.dependencies:56]: I had forgotten to do this, will do this now...

Comment (cgsunkel) [reply Dockerfile.dependencies:56]: Done now

**Review by oliverjwroberts — APPROVED** [review]

**Review by swenban — APPROVED** [review]

**Pull request merged**